pkg/hubble/server/serveroption/option.go

```diff
@@ -72,6 +72,8 @@ func WithUnixSocketListener(path string) Option {
 		}
 		if os.Getuid() == 0 {
 			if err := api.SetDefaultPermissions(socketPath); err != nil {
+				socket.Close()
```

Comment:
I think socket.Close() will calll syscall.Unlink internally if I am not wrong ?

Reply:
You are correct! https://github.com/golang/go/blob/a38a917aee626a9b9d5ce2b93964f586bf759ea0/src/net/unixsock_posix.go#L169-L187

I guess we can remove the unlink and just have a close?